testing-chocolatey/local-testing.md

```diff
@@ -0,0 +1,86 @@
+[vbox-download]: https://www.virtualbox.org/wiki/Downloads
+
+## Getting Started
+
+A `Vagrantfile` is provided for ease of use, which simplifies the necessary setup.
+If you would like to setup your own VMs or servers for the purposes of this testing, see the [Generic](#generic) instructions.
+
+Otherwise, follow the [Vagrant](#vagrant) instructions.
+
+Once your VMs or servers are ready, follow the instructions in [Running `ansible-test`](#running-ansible-test) to run through the collection tests.
+
+### Vagrant
+
+1. Download the nupkg containing the Chocolatey version you'd like to test.
+1. Place the nupkg in the `provision_files` folder.
+1. Run `vagrant up` to spin up the Ubuntu 'server' and Windows 10 'client' VMs.
+
+Continue on to the [Running `ansible-test`](#running-ansible-test) section.
+
+### Generic
+
+1. Before continuing, prepare a Windows VM and a Linux VM (these instructions assume you're using Ubuntu 18.10 or newer, and a Windows 10 client)
+1. Run the following commands on the Linux server to setup Ansible and install the Chocolatey collection:
+
+    ```sh
+    sudo apt-get update
+    sudo apt-get install python3-venv -y
+
+    python3 -m venv ~/ansible
+    source ~/ansible/bin/activate
+
+    pip3 install --upgrade wheel
+    pip3 install ansible pywinrm
+
+    ansible-galaxy collection install chocolatey.chocolatey
+    ```
+
+1. Copy the **nupkg** file containing the version of Chocolatey you'd like to test onto the Windows VM, under `C:\packages` (you will probably need to create this folder as well).
+1. Run the following commands in an administrative session of Windows PowerShell to install Chocolatey and update it to the version you've placed in `C:\packages`:
+
+    ```powershell
+    & ([scriptblock]::Create((Invoke-WebRequest -Uri "https://chocolatey.org/install.ps1" -UseBasicParsing)))
+    choco upgrade chocolatey -y -s C:\packages --pre
+    ```
+
+1. Follow the instructions from [this blog post](https://pureinfotech.com/set-static-ip-address-windows-10/) to assign the Windows VM the static IP address `10.0.0.11`
+   This IP address is hard-coded into the `vagrant-inventory.winrm` file found in the `~/.ansible/collections/ansible_collections/chocolatey/chocolatey/tests/integration/targets` folder.
+   If you prefer, you can modify that file in your Linux server to point to a different IP address instead.
+
+## Running `ansible-test`
+
+1. If using Vagrant, SSH into the Ansible server with the following command:
+
+    ```sh
+    vagrant ssh choco_ansible_server
+    ```
+
+   Otherwise, open your Linux VM directly and continue from the VM terminal.
+1. Run the chocolatey.chocolatey collection tests against the Windows client machine:
+
+    ```sh
+    source ~/ansible/bin/activate
+    cd ~/.ansible/collections/ansible_collections/chocolatey/chocolatey
+    ansible-test windows-integration -vvvv --inventory vagrant-inventory.winrm --requirements
+    ```
```

Comment:
What should I do now?  Can I run this ansible-test command again after installing another Chocolatey Beta version, or should I destroy the VM's and run the vagrant up again?

Reply:
At the moment, best to either destroy them and start over, or fully uninstall Chocolatey and install the other beta version.

In future, would be good to get the tests fully cleaned up, but I've noticed a couple of them don't fully restore the prior state on the test client, so there's a couple of tests that fail on a rerun as they expect a clean slate before starting.